src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs

```diff
@@ -78,8 +78,18 @@ public override Task Invoke(IOwinContext owinContext)
             {
                 return ProcessNegotiationRequest(owinContext, context);
             }
+            
+            // Trim any trailing slashes
+            var normalized = context.Request.LocalPath.TrimEnd('/');
+
+            if (normalized.EndsWith("/hubs", StringComparison.OrdinalIgnoreCase) ||
+                normalized.EndsWith("/js", StringComparison.OrdinalIgnoreCase))
+            {
+                return Next.Invoke(owinContext);
+            }
 
-            return Next.Invoke(owinContext);
+            context.Response.StatusCode = 404;
```

Comment:
any unknown path return 404? I think we should only handle signalr related URLs.. What if the customer uses the same app server to act as a normal web API?

Reply:
For AspNet, it seems there is always a mapping for path: 
https://github.com/Azure/azure-signalr/blob/4e234d39729bd16eb750ecc0d9859c1b2aa91847/src/Microsoft.Azure.SignalR.AspNet/OwinExtensions.cs#L62
So I think this middleware will only work for signalr path.